internal/server/iigointernal/orchestration.go

```diff
@@ -76,64 +91,86 @@ func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.C
-	judgeWithdrawSuccess := legislativeBranch.withdrawJudgeSalary(g)
-	speakerWithdrawSuccess := executiveBranch.withdrawSpeakerSalary(g)
+	// Pay salaries into budgets
 
-	// Handle the lack of resources
-	if !presidentWithdrawSuccess {
-		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
-		return false, "Could not run IIGO since President has no resources to spend"
-	}
-	if !judgeWithdrawSuccess {
-		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
-		return false, "Could not run IIGO since Judge has no resources to spend"
-	}
-	if !speakerWithdrawSuccess {
-		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
-		return false, "Could not run IIGO since Speaker has no resources to spend"
+	errorJudicial := judicialBranch.sendPresidentSalary()
+	errorLegislative := legislativeBranch.sendJudgeSalary()
+	errorExecutive := executiveBranch.sendSpeakerSalary()
+	// Throw error
+	if errorJudicial != nil || errorLegislative != nil || errorExecutive != nil {
+		return false, "Cannot pay IIGO salary"
 	}
 
-	// Pay salaries into budgets
-	judicialBranch.sendPresidentSalary(&executiveBranch)
-	legislativeBranch.sendJudgeSalary(&judicialBranch)
-	executiveBranch.sendSpeakerSalary(&legislativeBranch)
-
 	// 1 Judge actions - inspect history
 	_, _ = judicialBranch.inspectHistory(g.IIGOHistory)
 
 	// 2 President actions
-	resourceReports := map[shared.ClientID]shared.Resources{}
-	var aliveClientIds []shared.ClientID
-	for _, v := range rules.VariableMap[rules.IslandsAlive].Values {
-		aliveClientIds = append(aliveClientIds, shared.ClientID(int(v)))
-		resourceReports[shared.ClientID(int(v))] = iigoClients[shared.ClientID(int(v))].ResourceReport()
+	resourceReports := map[shared.ClientID]shared.ResourcesReport{}
+	aliveClientIds := []shared.ClientID{}
+	for clientID, clientGameState := range g.ClientInfos {
+		if clientGameState.LifeStatus != shared.Dead {
+			aliveClientIds = append(aliveClientIds, clientID)
+			resourceReports[clientID] = iigoClients[clientID].ResourceReport()
+		}
+	}
+
+	// Throw error if any of the actions returns error
+	insufficientBudget := executiveBranch.broadcastTaxation(resourceReports, aliveClientIds)
+	var ruleToVoteReturn shared.PresidentReturnContent
+
+	if insufficientBudget == nil {
+		insufficientBudget = executiveBranch.requestAllocationRequest(aliveClientIds)
+	}
+	if insufficientBudget == nil {
+		insufficientBudget = executiveBranch.replyAllocationRequest(g.CommonPool, aliveClientIds)
+	}
+	if insufficientBudget == nil {
+		insufficientBudget = executiveBranch.requestRuleProposal(aliveClientIds)
+	}
+	if insufficientBudget == nil {
+		ruleToVoteReturn, insufficientBudget = executiveBranch.getRuleForSpeaker()
+	}
+	if insufficientBudget != nil {
+		return false, "Common pool resources insufficient for executiveBranch actions"
```

Comment:
seems like a `try...catch` approach--don't: it's super hard to debug: do you know which function has errored if this returns an error?

Reply:
If I do `return false, insufficientBudget`. Then the error will indicate which function returned the error. 
I'm not sure what other approaches to use.  We only want to run the next line if the one before was successful. Would you have a return statement after every call?

Reply:
> Would you have a return statement after every call? 
you should, if the error is not nil

Reply:
Thanks, Updated